cmd/zpool/zpool_main.c

```diff
@@ -2471,61 +2474,587 @@ zpool_do_import(int argc, char **argv)
+
+	if (flags & VS_HISTO_M)
+		printf("%-*s", namewidth, histo_vdev_name);
+	else
+		printf("%*s", namewidth, "");
+
+	/* For each bit set in flags */
+	for (f = flags; f; f &= ~BIT(idx)) {
+		idx = ffsll(f) - 1;
+		if (!force_column_width)
+			column_width = default_column_width(cb, idx);
+
+		rw_column_width = (column_width * 2) + 2;
+
+		/* Print our top labels centered over "read  write" label. */
+		for (i = 0; iostat_top_labels[idx][i] != NULL; i++) {
+			text_start = (int) ((rw_column_width)/2 -
+					strlen(iostat_top_labels[idx][i])/2);
+			printf("  %*s%s%*s",
+				text_start, " ", iostat_top_labels[idx][i],
+				(int)(rw_column_width - text_start -
+				strlen(iostat_top_labels[idx][i])), " ");
+		}
+	}
+
+	printf("\n");
+
+	printf("%-*s", namewidth, flags & VS_HISTO_M ? "latency" :
+		cb->cb_vdev_names_count ? "vdev" : "pool");
+
+	for (f = flags; f; f &= ~BIT(idx)) {
+		idx = ffsll(f) - 1;
+		if (!force_column_width)
+			column_width = default_column_width(cb, idx);
+
+		for (i = 0; iostat_bottom_labels[idx][i] != NULL; i++)
+			printf("  %*s", (int) column_width,
+					iostat_bottom_labels[idx][i]);
+	}
 
-	for (i = 0; i < cb->cb_namewidth; i++)
-		(void) printf("-");
-	(void) printf("  -----  -----  -----  -----  -----  -----\n");
+	printf("\n");
+	__print_iostat_separator(cb, force_column_width);
 }
 
 static void
 print_iostat_header(iostat_cbdata_t *cb)
 {
-	(void) printf("%*s     capacity     operations    bandwidth\n",
-	    cb->cb_namewidth, "");
-	(void) printf("%-*s  alloc   free   read  write   read  write\n",
-	    cb->cb_namewidth, "pool");
-	print_iostat_separator(cb);
+	__print_iostat_header(cb, 0, NULL);
 }
 
+
 /*
  * Display a single statistic.
  */
 static void
-print_one_stat(uint64_t value)
+print_one_stat(uint64_t value, enum zfs_nicenum_format format,
+		unsigned int column_size, boolean_t scripted)
 {
 	char buf[64];
+	__zfs_nicenum(value, buf, sizeof (buf), format);
+
+	if (scripted)
+		printf("\t%s", buf);
+	else
+		printf("  %*s", column_size, buf);
+}
+
+/*
+ * Calculate the vdev stats
+ *
+ * Subtract oldvs from newvs, apply a scaling factor, and save the resulting
+ * stats into calcvs.
+ */
+static void
+calc_iostats(vdev_stat_t *oldvs, vdev_stat_t *newvs, vdev_stat_t *calcvs,
+		iostat_cbdata_t *cb)
+{
+	int i, j;
+	uint64_t tdelta = newvs->vs_timestamp - oldvs->vs_timestamp;
+	double scale;
+	memcpy(calcvs, newvs, sizeof (*calcvs));
+
+	if ((oldvs->vs_timestamp == 0) && (cb->cb_flags & VS_HISTO_M)) {
+		/*
+		 * If we specify printing histograms with no time interval, then
+		 * print the histogram numbers over the entire lifetime of the
+		 * vdev.
+		 */
+		scale = 1;
+	} else {
+		if (tdelta == 0)
+			scale = 1.0;
+		else
+			scale = (double)NANOSEC / tdelta;
+	}
+
+	for (j = 0; j < ARRAY_SIZE(calcvs->vs_queue_histo); j++)
+		for (i = 0; i < ARRAY_SIZE(calcvs->vs_queue_histo[0]); i++)
+			calcvs->vs_queue_histo[j][i] = scale *
+			(newvs->vs_queue_histo[j][i] -
+			oldvs->vs_queue_histo[j][i]);
```

Comment:
[style] The existing style is a 4 space indent to continue a line (not a tab).  You'll want to go back and apply this consistently throughout the patch.